src/main/scala/uk/gov/hmrc/mongo/ReactiveRepository.scala

```diff
@@ -87,7 +87,7 @@ abstract class ReactiveRepository[A <: Any, ID <: Any](collectionName: String,
     }
   }
 
-  def ensureIndexes(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
+  final def ensureIndexes(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
```

Comment:
Out of curiosity. Will it still be possible to programmatically change the definition of existing indexes if ensureIndexes cannot be overridden? An example of this use case is here: https://github.com/hmrc/mongo-caching/blob/master/src/main/scala/uk/gov/hmrc/cache/repository/TTLIndexing.scala